modules/apps/document-library/document-library-web/src/main/java/com/liferay/document/library/web/internal/display/context/DLAdminDisplayContext.java

```diff
@@ -708,9 +711,17 @@ else if (Objects.equals(getOrderByCol(), "size")) {
 								assetEntry.getClassName(),
 								DLFileEntryConstants.getClassName())) {
 
-							results.add(
+							FileEntry fileEntry =
 								DLAppLocalServiceUtil.getFileEntry(
-									assetEntry.getClassPK()));
+									assetEntry.getClassPK());
+
+							if ((folderId ==
```

Comment:
Hi @tototrinh 

We have a similar problem if we select an asset library as the root folder:

with the same steps as described on https://issues.liferay.com/browse/LPS-172528

add this

14. Add an Asset library and link to the site
15. Add documents to the Asset Library
16. On the D&M widget select the asset library home folder as the root folder
17. Select test's Category1 on Category Filter

**Actual result**
 FileA and FileB are shown. 

**Expected result**
Nothing is shown 


So I suggest this code similar to this in the if:
```

	if (_isAncestorFolder(folderId, fileEntry) ||
		((folderId ==
			DLFolderConstants.
				DEFAULT_PARENT_FOLDER_ID) &&
		 (fileEntry.getRepositoryId() ==
			 getRepositoryId()))) {

		results.add(fileEntry);
	}
```

Thnaks!